expression/builtin_time.go

```diff
@@ -2748,25 +2748,42 @@ func (c *secToTimeFunctionClass) getFunction(args []Expression, ctx context.Cont
 	if err := c.verifyArgs(args); err != nil {
 		return nil, errors.Trace(err)
 	}
-	sig := &builtinSecToTimeSig{newBaseBuiltinFunc(args, ctx)}
+	var retFlen, retFsp int
+	argType := args[0].GetType()
+	argEvalTp := fieldTp2EvalTp(argType)
+	if argEvalTp == tpString {
+		retFsp = types.UnspecifiedLength
```

Comment:
I wrote it as this to emphasis that `ret_decimal = -1 when arg_type = string` is not because that it is a special case, but because that when string is used in arg, its own decimal is -1 (See https://github.com/pingcap/tidb/pull/4236)